core/src/main/java/bio/terra/pearl/core/service/study/StudyService.java

```diff
@@ -45,13 +46,21 @@ public List<Study> findByPortalId(UUID portalId) {
 
     @Transactional
     public Study create(Study study) {
-        Study newStudy = dao.create(study);
-        study.getStudyEnvironments().forEach(studyEnv -> {
-            studyEnv.setStudyId(newStudy.getId());
-            StudyEnvironment newEnv = studyEnvironmentService.create(studyEnv);
-            newStudy.getStudyEnvironments().add(newEnv);
-        });
-        return newStudy;
+        try {
+            Study newStudy = dao.create(study);
+            study.getStudyEnvironments().forEach(studyEnv -> {
+                studyEnv.setStudyId(newStudy.getId());
+                StudyEnvironment newEnv = studyEnvironmentService.create(studyEnv);
+                newStudy.getStudyEnvironments().add(newEnv);
+            });
+            return newStudy;
+        } catch (Exception e) {
+            if (e.getCause() instanceof PSQLException && e.getMessage().contains("study_shortcode_key")) {
```

Comment:
Moved this here from StudyController to bring error handling closer to source of error.